src/tainting/Dataflow_tainting.ml

```diff
@@ -298,6 +318,27 @@ let top_level_sinks_in_nodes config flow =
              Seq.empty)
   |> Seq.fold_left (fun s x -> Top_sinks.add x s) Top_sinks.empty
 
+(* Checks whether the sink corresponds has the shape
+ *
+ *     patterns:
+ *     - pattern: <func>(<args>)
+ *     - focus-metavariable: $MVAR
+ *
+ * In which case we know that the function call itself is not the sink, but
+ * either the <func> or one (or more) of the <args>.
+ *)
+let is_func_sink_with_focus taint_sink =
```

Comment:
i surmise that a pattern like
```
patterns:
- pattern: [$X, ...]
- focus-metavariable: $X
```
would still flag false positives for something like `[ok1 if tainted else ok2]`, due to not being special-cased the same way as here. i am not in love with the fact that we have to do this special casing at all, though i recognize that this would be a valid solution for the problem you're trying to solve.

do you this may present an issue? will there be a great deal of boilerplate in the future needing to add in further special cases?

Reply:
Why the `Top_sinks` trick is done is explained (hopefully) in `NOTE "Top sinks"`. Given that,  I'm trying to cover the cases that cause most of the problems right now, while trying not to break anything else. It's true that there are other sinks using `focus-metavariable` that would not benefit from this, but I'm not too worried about it for now. If it's asked for, we can easily add it. But if more cases are requested then I agree we need to generalize this instead.

Reply:
Well, actually the case you wrote should be covered... this should only affect function calls.

Reply:
oh i know this PR only affects function calls, i just meant someone might write something like this:
https://semgrep.dev/s/B81R

and expect no findings

Reply:
I'll take a look if it's easy to add those cases too, although I haven't got any complain except for the function case.

Reply:
https://github.com/returntocorp/semgrep/pull/7466